tuples of slices for indexing multi-dimensional arrays?

Thanks to the super-fast feature-request response back in July, Pythran now supports specifying slice arguments (https://github.com/serge-sans-paille/pythran/issues/1326), and then using it within the function body as index specifiers for NumPy (which should help in forcing views instead of copies). 

In multiple dimensions, this would require passing tuples of slices. However, e.g. the following code:
```python
#pythran export fun(float64[:,:], (slice, slice))
def fun(psi, ij): 
  psi[ij] = 2 * psi[ij]
```
causes Pythran to choke and output:
```
...
/usr/local/lib/python3.7/dist-packages/pythran/pythonic/types/tuple.hpp:78:13: error: invalid cast from type ‘const value_type’ {aka ‘const {anonymous}::pythonic::types::slice’} to type ‘long int’
     return {(T)std::get<I>(a)...};
             ^~~~~~~~~~~~~~~~~
/usr/local/lib/python3.7/dist-packages/pythran/pythonic/types/tuple.hpp:78:33: error: could not convert ‘{<expression error>}’ from ‘<brace-enclosed initializer list>’ to ‘{anonymous}::pythonic::types::array_base<long int, 2, {anonymous}::pythonic::types::tuple_version>’
     return {(T)std::get<I>(a)...};
                                 ^
WARNING: Compilation error, trying hard to find its origin...
WARNING: Nop, I'm going to flood you with C++ errors!
CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
```

Any hints?
Thanks!

--- a/pythran/pythonic/include/types/numpy_texpr.hpp
+++ b/pythran/pythonic/include/types/numpy_texpr.hpp
@@ -139,15 +139,17 @@ namespace types
     operator[](F const &filter) const;
     auto operator[](long i) const -> decltype(this->fast(i));
     auto operator[](long i) -> decltype(this->fast(i));
-    auto operator[](array<long, value> const &indices)
-        -> decltype(arg[array<long, 2>{{indices[1], indices[0]}}])
+    template <class T>
+    auto operator[](array<T, value> const &indices)
+        -> decltype(arg[array<T, 2>{{indices[1], indices[0]}}])
     {
-      return arg[array<long, 2>{{indices[1], indices[0]}}];
+      return arg[array<T, 2>{{indices[1], indices[0]}}];
     }
-    auto operator[](array<long, value> const &indices) const
-        -> decltype(arg[array<long, 2>{{indices[1], indices[0]}}])
+    template <class T>
+    auto operator[](array<T, value> const &indices) const
+        -> decltype(arg[array<T, 2>{{indices[1], indices[0]}}])
     {
-      return arg[array<long, 2>{{indices[1], indices[0]}}];
+      return arg[array<T, 2>{{indices[1], indices[0]}}];
     }
     template <class T0, class T1>
     auto operator[](std::tuple<T0, T1> const &indices) -> decltype(

--- a/pythran/tests/test_ndarray.py
+++ b/pythran/tests/test_ndarray.py
@@ -1160,3 +1160,9 @@ def hanning(M):
                 B = x_transScores[x_transList]
                 return  x_transFrames[x_transList].astype(int),B'''
         self.run_test(code, numpy.arange(300.), numpy_vexpr_static_shape=[NDArray[float, :]])
+
+    def test_subscripting_slice_array_transpose(self):
+        code = 'def subscripting_slice_array_transpose(x): return x.T[(slice(0,1),slice(0,1))]'
+        self.run_test(code,
+                      numpy.arange(200.).reshape(10, 20),
+                      subscripting_slice_array_transpose=[NDArray[float, :, :]])

